### PR TITLE
[capi] add the capi e2es to the presubmits

### PR DESCRIPTION
**Pull request by chuckha**
Signed-off-by: Chuck Ha <chuckh@vmware.com>

/assign @detiber @wfernandes

Fixes kubernetes-sigs/cluster-api#1884

### CODE REVIEW

File: config/jobs/kubernetes-sigs/cluster-api/cluster-api-presubmits.yaml [hunk 1]
```diff
@@ -104,6 +104,29 @@ presubmits:
     annotations:
       testgrid-dashboards: sig-cluster-lifecycle-cluster-api
       testgrid-tab-name: pr-integration
+  - name: pull-cluster-api-e2e
```
Comment (detiber) [inline config/jobs/kubernetes-sigs/cluster-api/cluster-api-presubmits.yaml]: Should we also add a periodic (and possibly also a postsubmit) job for this as well?

Comment (chuckha) [reply config/jobs/kubernetes-sigs/cluster-api/cluster-api-presubmits.yaml]: this e2e tests that machine CRD and machine webhook is validating correctly -- I don't think we'd need a post submit or periodic for this. why do you think we need those?

**Comment by wfernandes** [comment]
/lgtm
I'm fine with having them run at the same cadence as the capd tests for now.

**Comment by chuckha** [comment]
Ah, the capd tests aren't running as a pull job, they're running as a periodic and as an optional PR job that you can trigger if you wish

File: config/jobs/kubernetes-sigs/cluster-api/cluster-api-presubmits.yaml (hunk 1 above)
Comment (detiber) [reply config/jobs/kubernetes-sigs/cluster-api/cluster-api-presubmits.yaml]: I'd like to have it as a signal that the current state of HEAD (or a particular release branch) is in a good state for release signal.

Comment (chuckha) [reply config/jobs/kubernetes-sigs/cluster-api/cluster-api-presubmits.yaml]: sure, no problem, one sec

**Comment by chuckha** [comment]
@detiber Updated! Ptal!

**Comment by chuckha** [comment]
my fault, really thought i had checked all the possible duplicate values. Fixed now.

**Pull request merged**